### PR TITLE
Update kv certs sync stack

### PR DESCRIPTION
**Pull request by g2vinay**
Integrate Sync Stack into KV Certificates

Paging and Poller APIs will be updated later, once support for them lands in azure-core.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTest.java [hunk 1]
```diff
@@ -61,7 +65,8 @@ private void createCertificateClient(HttpClient httpClient, CertificateServiceVe
 
     private void createCertificateClient(HttpClient httpClient, CertificateServiceVersion serviceVersion,
                                          String testTenantId) {
-        HttpPipeline httpPipeline = getHttpPipeline(httpClient, testTenantId);
+        HttpPipeline httpPipeline = getHttpPipeline(buildSyncAssertingClient(httpClient == null
```
Comment (kasobol-msft) [inline sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTest.java:68]: we should create tests for async client now.

**Review by kasobol-msft — COMMENTED** [review]
lgtm.

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java [hunk 2]
```diff
@@ -93,45 +65,26 @@
  * @see CertificateClientBuilder
  * @see PagedFlux
  */
-@ServiceClient(builder = CertificateClientBuilder.class, isAsync = true, serviceInterfaces = CertificateService.class)
+@ServiceClient(builder = CertificateClientBuilder.class, isAsync = true, serviceInterfaces = CertificateClientImpl.CertificateService.class)
```
Comment (mssfang) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java:68]: CertificateClientImpl is in the implementation folder. Does it allow to be explored?

Comment (mssfang) [reply sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java:68]: nvm

**Review by mssfang — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificateClientImpl.java [hunk 3]
```diff
@@ -0,0 +1,1771 @@
...
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
+    }
+
+    private Mono<Response<CertificateOperation>> createCertificateWithResponseAsync(String certificateName, CertificatePolicy certificatePolicy, boolean enabled, Map<String, String> tags, Context context) {
+        CertificateRequestParameters certificateRequestParameters = new CertificateRequestParameters()
+            .certificatePolicy(new CertificatePolicyRequest(certificatePolicy))
+            .certificateAttributes(new CertificateRequestAttributes().enabled(enabled))
+            .tags(tags);
+        return service.createCertificateAsync(vaultUrl, certificateName, apiVersion, ACCEPT_LANGUAGE, certificateRequestParameters, CONTENT_TYPE_HEADER_VALUE,
+            context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificateClientImpl.java:997]: I noticed we didn't have logging statements for this operation before, but I think we should:
```suggestion
        return service.createCertificateAsync(vaultUrl, certificateName, apiVersion, ACCEPT_LANGUAGE, certificateRequestParameters, CONTENT_TYPE_HEADER_VALUE,
            context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
            .doOnRequest(ignored -> logger.verbose("Starting creation of certificate - {}",  certificateName))
            .doOnSuccess(response -> logger.verbose("Started creation of certificate - {}", certificateName))
            .doOnError(error -> logger.warning("Failed to create the certificate - {}", certificateName, error));
```

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateClient.java [hunk 4]
```diff
@@ -57,25 +57,25 @@
  * @see CertificateClientBuilder
  * @see PagedIterable
  */
-@ServiceClient(builder = CertificateClientBuilder.class, serviceInterfaces = CertificateService.class)
+@ServiceClient(builder = CertificateClientBuilder.class, serviceInterfaces = CertificateClientImpl.CertificateService.class)
 public final class CertificateClient {
-    private final CertificateAsyncClient client;
+    private final CertificateClientImpl certificateClient;
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateClient.java:62]: I think `implClient` would be a better fitting name for this variable.

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java [hunk 5]
```diff
@@ -93,45 +65,26 @@
...
-    private static final String KEYVAULT_TRACING_NAMESPACE_VALUE = "Microsoft.KeyVault";
-
-    private static final Duration DEFAULT_POLLING_INTERVAL = Duration.ofSeconds(1);
-
-    private final String vaultUrl;
-    private final CertificateService service;
     private final ClientLogger logger = new ClientLogger(CertificateAsyncClient.class);
-
-    private final HttpPipeline pipeline;
+    private final CertificateClientImpl certificateClient;
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java:71]: Same suggestion as in the sync client: I think this should better be named `implClient`.

**Review by vcolin7 — COMMENTED** [review]
Things look good for the most part, we just need to add logging statements for a few async operations. I also added a naming suggestion for the inner clients in both `CertificateClient` and `CertificateAsyncClient`.

**Pull request merged**